### ISSUE TITLE
is it support typescript ?

### ISSUE THREAD

**Issue by Liaohuanle** [CookPete/react-player#299]
is it support typescript ?

**Comment by narkowicz** [comment]
Yes, type definitions are bundled with the package as of 5f48260626e752550ba381f3c8ba8d0545c1259f

**Comment by CookPete** [comment]
Yeah, ReactPlayer should work fine with TypeScript.

**Comment by jp3492** [comment]
Im curios on how i add the typescript config for ReactPlayer. Im using create-react-app and i cant figure out how to solve for this. any suggestions?

**Comment by narkowicz** [comment]
@jp3492 the typescript definitions are included in the npm package and referenced in `package.json` under the `"typings"` field so in theory there's nothing to do other than install the package. I'm not a contributor but in our experience the type defs _just work™_.

I tried out adding `react-player` to a fresh `create-react-app` (`v2.1.3` with `--typescript` flag) and there were no issues with the `react-player` type defs which imported fine with the module.

**However** - I did run into an issue when installing `react-player` with `npm` after `create-react-app` was initialised with `yarn` (not directly but via `npx` 🤦‍♂️ )!

For some reason this removed the `@types/react` defs from `node_modules` (obviously a dependency of `react-player` types and `create-react-app`). Running `npm i` resolved this, but probably better to stick with `yarn add` if using `yarn`. Hope that helps?
